harness/determined/tensorboard/base.py

```diff
@@ -111,22 +154,25 @@ def __init__(
 
     def run(self) -> None:
         while True:
-            file_paths = self._work_queue.get()
+            path_info_list = self._work_queue.get()
 
             # None is the sentinel value
             # to signal the thread to exit
-            if file_paths is None:
+            if path_info_list is None:
                 return
 
             # Try-catch is used to avoid exception from
             # one failed sync attempt to cause the thread to exit.
             try:
-                self._upload_function(file_paths)
+                self._upload_function(path_info_list)
             except Exception as e:
                 logging.warning(f"Sync of Tensorboard files failed with error: {e}")
 
-    def upload(self, paths: List[pathlib.Path]) -> None:
-        self._work_queue.put(paths)
+    def upload(self, path_info_list: List[PathUploadInfo]) -> None:
+        self._work_queue.put(path_info_list)
 
     def close(self) -> None:
         self._work_queue.put(None)
+        while self.is_alive():
+            logging.info("Waiting for Tensorboard files to finish uploading")
+            self.join(10)
```

Comment:
optional: I was initially thinking that it would be preferable to only emit this if we noticed we were waiting for some initial period of time, perhaps like this:

```python
while not self.join(10):
    logging.info("Waiting for Tensorboard files to finish uploading")
```

But I kind of like the subtle advertisement that "yeah, we just uploaded your tensorboard metrics in the background".  So either way is fine.